lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -863,20 +874,29 @@ static LogicalResult verifyIfOp(IfOp ifOp) {
 static LogicalResult verifyWhileOp(WhileOp whileOp) {
   auto component = whileOp->getParentOfType<ComponentOp>();
   auto wiresOp = component.getWiresOp();
-  auto groupName = whileOp.groupName();
-  auto groupOp = wiresOp.lookupSymbol<GroupOp>(groupName);
 
+  if (whileOp.body().front().empty())
+    return whileOp.emitError() << "empty body region.";
+
+  auto optGroupName = whileOp.groupName();
```

Comment:
Nit: Let's try to to avoid `auto` unless it is extremely long (e.g. iterators) or extremely obvious (e.g. `cast<MyType>()`). It makes the code more friendly for new Calyx users and future contributors. Let me know if you disagree.

Reply:
I've come to pretty much using `auto` for everything but built-in types, which (to me) also seems to be the general approach in much of the MLIR codebase. However, in the given case, i agree that `Optional<StringRef>` here would aid in being explicit about what is going on.

Reply:
https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

Again, I find that `auto` can obfuscate code. Something that is obvious to us may not be obvious to first-time readers. I don't imagine the type of say, `groupName`, will change any time soon, but perhaps I'm wrong.